### DISCUSSION TITLE
Is it possible to dump the entire request (to the response)?

### DISCUSSION THREAD

**Discussion by cskwrd** [rwf2/Rocket#2384] [Questions] [answered]
I'm trying to implement a request inspector. Can a person dump or otherwise echo the (raw) incoming request back to the client? 

I have attempted this so far, by reconstructing the request as best I can using the data and functions available in a fairing. I can get most of the way there, but not having access to the raw request results in a few minor changes from the actual request sent.

**Comment by the10thWiz** [comment, marked as answer]
The basic answer is no, Rocket doesn't support this. This is a consequence of Rocket's focus on correctness and safety; Rocket handles the parsing of untrusted requests, and your code only sees well formed requests. In this case, you would likely be better served by a separate tool that can intercept and parse requests independently of the back-end server. Wireshark & Burp suite are quite popular choices, as well as the Firefox & Chrome devtools also provide quite good viewers for requests & responses.

**Reply by cskwrd** [reply]
I thought that might be the case. Thank you for the response!
